### PR TITLE
HARP-5644: Support for Political Views via FeatureModifiers stack.

### PR DESCRIPTION
**Pull request by kkostecki**
Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: @here/harp-mapview/lib/DataSource.ts [hunk 1]
```diff
@@ -398,6 +398,17 @@ export abstract class DataSource extends THREE.EventDispatcher {
         // to be overloaded by subclasses
     }
 
+    /**
+     * Used to express different country point of view (political view).
+     *
+     * @note Set to null if you want to reset to default point of view.
```
Comment (zbigg) [inline @here/harp-mapview/lib/DataSource.ts:404]: `null` -> `undefined`

File: @here/harp-mapview/lib/MapView.ts [hunk 2]
```diff
@@ -1463,6 +1477,36 @@ export class MapView extends THREE.EventDispatcher {
...
+     */
+    set politicalView(pov: string | undefined) {
+        if (this.m_politicalView === pov) {
+            return;
+        }
+        this.m_politicalView = pov;
+        this.m_tileDataSources.forEach((dataSource: DataSource) => {
+            dataSource.setPoliticalView(pov);
+        });
+        this.clearTileCache();
```
Comment (zbigg) [inline @here/harp-mapview/lib/MapView.ts:1506]: Already commented.
Not needed here.
It's DataSource responsibility to update it's Tiles in implementation of DataSource.setPoliticalView. For some DS no invalidation is needed.

File: @here/harp-mapview/lib/MapView.ts [hunk 3]
```diff
@@ -1463,6 +1477,36 @@ export class MapView extends THREE.EventDispatcher {
...
+        this.update();
```
Comment (zbigg) [inline @here/harp-mapview/lib/MapView.ts:1507]: Not needed too

File: @here/harp-mapview/lib/MapView.ts [hunk 4]
```diff
@@ -548,6 +548,18 @@ export interface MapViewOptions extends TextElementsRendererOptions, Partial<Loo
...
     languages?: string[];
 
+    /**
+     * Sets the data sources to use specific country point of view (political view).
+     *
+     * This option may result in rendering different country borders then commonly accepted for
+     * some regions and it mainly regards to so called __disputed borders__. Although not all
+     * data sources or themes may support it.
+     *
+     * @note Country code should be coded in ISO 3166-1 alpha-2 standard, if this option
```
Comment (zbigg) [inline @here/harp-mapview/lib/MapView.ts:558]: `ISO 3166-1 alpha-2` (https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) uses upper case codes everywhere.

This document: https://github.com/tilezen/vector-datasource/blob/master/docs/layers.md says "lower-case ISO 3166-1 alpha-2 codes or the pseudo-code iso" ...

File: @here/harp-omv-datasource/lib/OmvDataSource.ts [hunk 5]
```diff
@@ -273,6 +292,18 @@ export class OmvDataSource extends TileDataSource<OmvTile> {
...
     }
 
+    /** @override */
+    setPoliticalView(politicalView?: string): void {
+        if (this.m_decoderOptions.politicalView !== politicalView) {
+            this.m_decoderOptions.politicalView = politicalView;
+            this.configureDecoder(undefined, undefined, undefined, {
+                filterDescription: this.m_decoderOptions.filterDescription,
+                featureModifiers: this.m_decoderOptions.featureModifiers,
+                politicalView
```
Comment (zbigg) [inline @here/harp-omv-datasource/lib/OmvDataSource.ts:302]: `undefined` will not be transfered, so "unset doesn't work" here.

Yes, i KNOW `null` would work here, but we just need some workaround like `"neutral"` text and OmvDecoder should be able to intrpret it.

File: @here/harp-mapview/lib/MapView.ts (hunk 4 above)
Comment (kkostecki) [reply @here/harp-mapview/lib/MapView.ts:558]: Ok, good catch, I will update the documentation that it is lowercase.

File: @here/harp-mapview/lib/DataSource.ts (hunk 1 above)
Comment (kkostecki) [reply @here/harp-mapview/lib/DataSource.ts:404]: Ok.

File: @here/harp-mapview/lib/MapView.ts (hunk 2 above)
Comment (kkostecki) [reply @here/harp-mapview/lib/MapView.ts]: Ok removed.

File: @here/harp-mapview/lib/MapView.ts (hunk 3 above)
Comment (kkostecki) [reply @here/harp-mapview/lib/MapView.ts]: Done.

File: @here/harp-omv-datasource/lib/OmvDataSource.ts (hunk 5 above)
Comment (kkostecki) [reply @here/harp-omv-datasource/lib/OmvDataSource.ts]: Ok solved with empty string for default point of view - no OmvPoliticalViewFeatureModifier.

**Comment by kkostecki** [comment]
@harpgl-bot retest this please

**Comment by kkostecki** [comment]
@harpgl-bot retest this please

**Pull request merged**